neurom/core/_soma.py

```diff
@@ -134,19 +207,53 @@ def center(self):
         return np.mean(points[:, COLS.XYZ], axis=0)
 
     def __str__(self):
-        return 'SomaSimpleContour(%s) <center: %s, radius: %s>' % \
-            (repr(self._points), self.center, self.radius)
+        return ('SomaSimpleContour(%s) <center: %s, radius: %s>' %
+                (repr(self._points), self.center, self.radius))
+
 
+def _get_type(points, soma_class):
+    '''get the type of the soma
 
-def make_soma(points, soma_check=None):
+    Args:
+        points: Soma points
+        soma_class(str): one of 'contour' or 'cylinder' to specify the type
+    '''
+    assert soma_class in ('contour', 'cylinder', )
+    npoints = len(points)
+    if soma_class == 'contour':
+        return {0: None,
+                1: SomaSinglePoint,
+                3: SomaThreePoint,
+                2: None}.get(npoints, SomaSimpleContour)
+    elif soma_class == 'cylinder':
+        if(npoints == 3 and
+           points[0][COLS.P] == -1 and
+           points[1][COLS.P] == 1 and
+           points[2][COLS.P] == 1):
+            L.warning('Using neuromorpho 3-Point soma')
+            # NeuroMorpho is the main provider of morphologies, but they
+            # with SWC as their default file format: they convert all
+            # uploads to SWC.  In the process of conversion, they turn all
+            # somas into their custom 'Three-point soma representation':
+            #  http://neuromorpho.org/SomaFormat.html
+
+            return SomaThreePointCylinders
+
+        return {0: None,
```

Comment:
Seems easier to just use the dict lookup on point count, this way if we have something that has a 2 point soma, it's easy to find, for instance